server/gc/src/main/java/org/apache/accumulo/gc/GarbageCollectionAlgorithm.java

```diff
@@ -153,10 +153,10 @@ private void removeCandidatesInUse(GarbageCollectionEnvironment gce,
       tableIdsSeen.add(ref.getTableId());
 
       if (ref.isDirectory()) {
-        var dirReference = (ReferenceDirectory) ref;
-        ServerColumnFamily.validateDirCol(dirReference.getTabletDir());
+        var dirReference = (ReferenceFile) ref;
+        ServerColumnFamily.validateDirCol(dirReference.getMetadataEntry());
 
-        String dir = "/" + dirReference.tableId + "/" + dirReference.getTabletDir();
+        String dir = "/" + dirReference.tableId + "/" + dirReference.getMetadataEntry();
```

Comment:
The `dirReference` var was only created because `getTabletDir()` needed to be called. 
Since all references now use the Reference interface, the recast variable can be removed
```
       ServerColumnFamily.validateDirCol(ref.getMetadataEntry());

        String dir = "/" + ref.tableId + "/" + ref.getMetadataEntry();
```

Reply:
Done.